### ISSUE TITLE
New admin/client certs are not created if passing in existing cert for apache2.

### ISSUE THREAD

**Issue by grigutis** [logicnow/BlueSky#43]
I'm running the docker build and if I pass in an SSL cert/key for Apache (SSL_CERT/SSL_KEY), then the certs needed for Bluesky admin and client are not created.

The problem occurs in client-config.sh lines 40–52:

	if [ "$(find /certs -maxdepth 1 -type f -not -path '*/\.*')" ]; then
		# fixup files
		echo "Putting the provided ssh keys in place..."
		ln -fs /certs/blueskyclient.key /usr/local/bin/BlueSky/Server/
		ln -fs /certs/blueskyclient.pub /usr/local/bin/BlueSky/Client/
		ln -fs /certs/blueskyadmin.key /usr/local/bin/BlueSky/Server/
		ln -fs /certs/blueskyadmin.pub /usr/local/bin/BlueSky/Admin\ Tools/
		ln -fs /certs/blueskyd.pub /usr/local/bin/BlueSky/Server/
		ln -fs /certs/blueskyd /usr/local/bin/BlueSky/Server/
	else
		echo "fresh docker container - lets rebuild keys..."
		IN_DOCKER_FRESH=true
	fi

Basically, any file in the certs directory satisfies this find, so `IN_DOCKER_FRESH` is never set to true.

One possible solution would be to add `-name 'bluesky*'` to that find, but that doesn't take into account that my apache cert might be named "bluesky_ssl.pem". Another would be to add variables to explicitly specify the existing bluesky keys. Or maybe there's another, better solution.

**Comment by sphen13** [comment]
Ah great. Yeah originally I had the ssl certs passed to a different location but then I simplified it later on. This is for sure a bug. Thanks for the suggestions.

**Comment by sphen13** [comment]
Ugh i feel like i have been staring at this. The way we are checking right now is not too slick - just seeing if there is anything in that folder and assuming its all good if there is something.  we really should check for each and every one and rekey if necessary... you think?

**Comment by sphen13** [comment]
@grigutis not sure if you are willing try try - but i think this should work appropriately 
https://github.com/logicnow/BlueSky/commit/798eaae05377b70718f65bbaaa94e41ec6645d0b

**Comment by grigutis** [comment]
Thanks Stephen! Yes, I'm willing to test it in a separate container, but how would I specify the docker_clientconfig_fix branch in my initial docker run command? It doesn't look like that's possible from my searching. Is there some other way to replace that file in the container before the initial run?

btw, I worked around this in the meantime by using symbolic links to the cert/key instead of using copies of the actual files.

**Comment by sphen13** [comment]
thanks @grigutis - i have pushed a new build to docker hub.  you will need to refer to it as sphen/bluesky:dev in your docker run command.  if you have ever pulled dev before, pull it first: `docker pull sphen/bluesky:dev`

**Comment by grigutis** [comment]
This looks good to me. All the certs are now created even with my existing SSL cert/key:

    root@a6160b121525:/certs# ls -al
    total 56
    drwxrwxrwx 2 root     root 4096 Sep 16 09:14 .
    drwxr-xr-x 1 root     root 4096 Sep 16 09:14 ..
    -rw-r--r-- 1 root     root 1708 Sep 16 09:14 blueskyadmin.key
    -rw-r--r-- 1 root     root 1042 Sep 16 09:14 blueskyadmin.pub
    -rw-r--r-- 1 root     root 1704 Sep 16 09:14 blueskyclient.key
    -rw-r--r-- 1 root     root 1042 Sep 16 09:14 blueskyclient.pub
    -rw------- 1 www-data root 1675 Sep 16 09:14 blueskyd
    -rw-r--r-- 1 root     root  407 Sep 16 09:14 blueskyd.pub
    -rw-r--r-- 1 root     root 3830 Sep 16 09:13 fullchain.pem
    -rw-r--r-- 1 root     root 1704 Sep 16 09:12 privkey.pem
    -rw------- 1 root     root  399 Sep 16 09:14 ssh_host_ed25519_key
    -rw-r--r-- 1 root     root   91 Sep 16 09:14 ssh_host_ed25519_key.pub
    -rw------- 1 root     root 1675 Sep 16 09:14 ssh_host_rsa_key
    -rw-r--r-- 1 root     root  391 Sep 16 09:14 ssh_host_rsa_key.pub

👍
